From 15db697c5666b6e40545bd75e30f470ccf973caf Mon Sep 17 00:00:00 2001
From: Peng Liu <liupeng256@huawei.com>
Date: Sat, 21 May 2022 12:18:57 +0800
Subject: [PATCH] Documentation: arm64: Add limitation for exactmap
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 15db697c5666b6e40545bd75e30f470ccf973caf
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56F4Z
CVE: NA

--------------------------------

For arm64, the exactmap has to remove all memblock and then add
the user assigned range of memblock. EFI initializes memblock too
late so removing all memblock after it will cause panic. Hence,
the exactmap can be just used in dt boot mode, add this limitation
to Documentation/admin-guide/kernel-parameters.txt.

Signed-off-by: Peng Liu <liupeng256@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 Documentation/admin-guide/kernel-parameters.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/admin-guide/kernel-parameters.txt b/Documentation/admin-guide/kernel-parameters.txt
index 4c9edd6b4729..a70ca42d3251 100644
--- a/Documentation/admin-guide/kernel-parameters.txt
+++ b/Documentation/admin-guide/kernel-parameters.txt
@@ -2813,6 +2813,8 @@
 
 	memmap=exactmap	[KNL,X86,ARM64] Enable setting of an exact
 			E820 and ARM64 memory map, as specified by the user.
+			For ARM64, this setting is limited to dt boot mode as
+			exact mapping must be done after initializing memblock.
 			Such memmap=exactmap lines can be constructed based on
 			BIOS output or other requirements. See the memmap=nn@ss
 			option description.
-- 
2.34.1

